### PR TITLE
Fix a NPE when cache is unavailable. Start polling on dynamic attribute event if created after initDevice()

### PR DESCRIPTION
**Pull request by Pascal-Verdier**
Fix a NPE when cache is unavailable.
Start polling on dynamic attribute event if created after initDevice()

### CODE REVIEW

File: server/src/main/java/org/tango/server/cache/TangoCacheManager.java [hunk 1]
```diff
@@ -400,6 +401,8 @@ public synchronized SelfPopulatingCache getAttributeCache(final AttributeImpl at
             AttributeCache attrCache = attributeCacheMap.get(attr);
             if (attrCache == null) {
                 attrCache = extTrigAttributeCacheMap.get(attr);
+                if (attrCache == null)
```
Comment (Ingvord) [inline server/src/main/java/org/tango/server/cache/TangoCacheManager.java:404]: Too many nested ifs. I will move it into a separate method.

File: server/src/main/java/org/tango/server/servant/AttributeGetterSetter.java [hunk 2]
```diff
@@ -231,6 +231,10 @@ static void setAttributeValue(final fr.esrf.Tango.AttributeValue[] values, final
                 if (e.getCause() instanceof DevFailed) {
                     back[i] = TangoIDLAttributeUtil.toAttributeValue5Error(names[i], AttrDataFormat.FMT_UNKNOWN, 0,
                             (DevFailed) e.getCause());
+                } else {
```
Comment (Ingvord) [inline server/src/main/java/org/tango/server/servant/AttributeGetterSetter.java:234]: Probably we do not need this if here. If *CacheException* is a *DevFailed* we just need 2 **catch** clauses here:

```java
catch(CacheException e){...}
catch(DevFailed e){...}

```

**Review by Ingvord — APPROVED** [review]
Looks fine to me. I have pointed out a few minor issues, I will fix them by myself. 

Just a general comment - it is better to avoid PR with more than one changes, especially if it is a small fixes. Prefer one change - one PR strategy, then it is much easier to manipulate them, for instance if one of the changes has to be reverted

File: server/src/main/java/org/tango/server/cache/TangoCacheManager.java (hunk 1 above)
Comment (Pascal-Verdier) [reply server/src/main/java/org/tango/server/cache/TangoCacheManager.java:404]: True. But my goal was to fix the problem quickly and not to change the code itself.

File: server/src/main/java/org/tango/server/servant/AttributeGetterSetter.java (hunk 2 above)
Comment (Pascal-Verdier) [reply server/src/main/java/org/tango/server/servant/AttributeGetterSetter.java:234]: If you want.

**Comment by Ingvord** [comment]
I have made a few changes before merging this PR. Namely, a new exception class has been added - *NoCacheFoundException*. It is being thrown by TangoCacheManager#getAttributeCache method. This way this it is clearer to handle situations when cache is not found.

**Pull request merged**